src/components/Traceroute/Traceroute.tsx

```diff
@@ -55,89 +43,206 @@ const TracerouteTool = () => {
+    );
+
+    //Traceroute Options
+    const tracerouteSwitch = [
+        "Traceroute ICMP scan",
+        "Traceroute TCP scan",
+        "Traceroute UDP scan",
+        "Traceroute custom scan",
+    ]; // Options for different types of traceroute scans
+
+    /**
+     * onSubmit: Asynchronous handler for the form submission event event related to executing traceroute commands.
+      It constructs command arguments, executes them via a shell script, and updates the UI with the results or errors.
+     *
+     * @param {FormValuesType} values - Contains the type of traceroute scan (ICMP, TCP, UDP, custom),
+     * the destination hostname, and optional custom traceroute options.
+     */
     const onSubmit = async (values: FormValuesType) => {
         let args = [``];
 
-        //Switch case
+        // Switch case to handle different traceroute scan options based on user selection.
         switch (values.tracerouteSwitch) {
-            case "Traceroute ICMP scan": //traceroute syntax: traceroute -I <hostname>
+            case "Traceroute ICMP scan":
+                // Traceroute ICMP scan uses the '-I' option
+                // Syntax: traceroute -I <hostname>
                 args = [`/usr/share/ddt/Bash-Scripts/Tracerouteshell.sh`];
                 args.push(`-I`);
-                args.push(`${values.hostname}`);
+                args.push(`${values.hostname}`); // Adds the hostname to the arguments list.
                 try {
+                    // Executed the traceroute command using a shell script and captures the output.
                     let output = await CommandHelper.runCommand("bash", args);
-                    setOutput(output);
+                    setOutput(output); // Sets the output to the state.
+                    setAllowSave(true);
                 } catch (e: any) {
-                    setOutput(e);
+                    setOutput(e); // Sets error to the state if command fails.
                 }
 
                 break;
 
-            case "Traceroute TCP scan": //traceroute syntax: traceroute -T <hostname>
+            case "Traceroute TCP scan":
+                // TRaceroute TCP scan uses the '-T' option.
+                // Syntax: traceroute -T <hostname>
                 args = [`/usr/share/ddt/Bash-Scripts/Tracerouteshell.sh`];
                 args.push(`-T`);
-                args.push(`${values.hostname}`);
+                args.push(`${values.hostname}`); // Adds the hostname to the arguments list.
                 try {
                     let output = await CommandHelper.runCommand("bash", args);
-                    setOutput(output);
+                    setOutput(output); //Sets the output to the state.
+                    setAllowSave(true);
                 } catch (e: any) {
-                    setOutput(e);
+                    setOutput(e); //Sets error to the state if command fails.
                 }
 
                 break;
```

Comment:
Review requirement for a break this is likely not required in switch stack.

Reply:
I tested the break statement and I believe it is required as it negatively impacts the result and execution providing incorrect executions.